### DISCUSSION TITLE
Exclude specific event type

### DISCUSSION THREAD

**Discussion by exoRift** [community/community#25410] [Actions]
<p>I want to exclude the <code>prerelease</code> event type for releases. How do I go about doing that?</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/exorift">@exorift</a>,</p>
<p>You can limit your workflow runs to specific activity types using the <strong>types</strong> keyword.<br>
In your case, you can configure the event in the workflow file like as this:</p>
<pre><code class="lang-yaml">on:
  release:
    types: [prereleased]
</code></pre>
<p>With this configuration, the workflow will be triggered only when you set the release as a <strong>pre-release</strong>.<br>
To view more details, you can see the “<a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#release" rel="noopener nofollow ugc"><strong>release event</strong></a>”.</p>

**Comment by exoRift** [comment]
<p>I already have <code>created</code> under types, but that includes prereleases. I want to exclude prereleases from executing the action</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/exorift">@exorift</a>,</p>
<blockquote>
<p>I already have <strong>created</strong> under types, but that includes prereleases.</p>
</blockquote>
<p>No, <strong>created</strong> does not includes <strong>prereleases</strong>. You may have some misunderstandings here.</p>
<p>When you click the “<strong>Draft a new release</strong>” button to create a new release and publish it as a <strong>pre-release</strong>, actually this operation includes two activity types of <strong>release</strong> event.</p>
<ul>
<li>
<p><strong>created</strong> - - A new release is created and saved. This activity will be triggered after you click the “<strong>Draft a new release</strong>” button and the “<strong>Publish release</strong>” button.</p>
</li>
<li>
<p><strong>prereleases</strong> - - A release is set as a <strong>pre-release</strong> and saved. This activity will be triggered after you select the “<strong>This is a pre-release</strong>” option and click the “<strong>Publish release</strong>” (or “<strong>Update release</strong>”) button.</p>
</li>
</ul>
<p>When you click the “<strong>Draft a new release</strong>” button, then select the “<strong>This is a pre-release</strong>” option, and click the “<strong>Publish release</strong>” button:</p>
<ul>
<li>
<p>If you configure the event in the workflow file like as this:</p>
<pre><code class="lang-yaml">on:
  release:
    types: [created, prereleased]
</code></pre>
<p>The workflow will be triggered twice by different activity types, one for <strong>created</strong> and another for <strong>prereleased</strong>.<br>
</p>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/5/a503142b3fb43fb1c7010459894fdc48328f205d.png" data-download-href="/uploads/short-url/nxLnusbtdGWTeqTGbZQ4p1WyzO5.png?dl=1" title="example-01" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/46307996/181090999-edb2bbd9-dbe0-4ac2-b567-301e6cc365c9.png" alt="example-01" data-base62-sha1="nxLnusbtdGWTeqTGbZQ4p1WyzO5" width="690" height="283" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/5/a503142b3fb43fb1c7010459894fdc48328f205d_2_690x283.png, https://aws1.discourse-cdn.com/github/original/3X/a/5/a503142b3fb43fb1c7010459894fdc48328f205d.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/a/5/a503142b3fb43fb1c7010459894fdc48328f205d.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/5/a503142b3fb43fb1c7010459894fdc48328f205d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">example-01</span><span class="informations">922×379 15.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
</li>
<li>
<p>If you configure the event in the workflow file like as this:</p>
<pre><code class="lang-yaml">on:
  release:
    types: [created]
</code></pre>
<p>The workflow will be triggered only once by the activity type <strong>created</strong>.<br>
</p>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/2/d2670b389be7f5d1c0ac2890134dd90d5bc66d60.png" data-download-href="/uploads/short-url/u1j2qDr53C13HVYD3YgfqIhD3VK.png?dl=1" title="example-02" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/46307996/181091000-17a30988-5add-4ea7-a1b1-cc079c80c48d.png" alt="example-02" data-base62-sha1="u1j2qDr53C13HVYD3YgfqIhD3VK" width="690" height="283" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/2/d2670b389be7f5d1c0ac2890134dd90d5bc66d60_2_690x283.png, https://aws1.discourse-cdn.com/github/original/3X/d/2/d2670b389be7f5d1c0ac2890134dd90d5bc66d60.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/d/2/d2670b389be7f5d1c0ac2890134dd90d5bc66d60.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/2/d2670b389be7f5d1c0ac2890134dd90d5bc66d60_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">example-02</span><span class="informations">922×379 15.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
</li>
</ul>
<p>More details, you can reference to <a href="https://github.com/TestWorkflowsRML/Event-Triggers/actions?query=workflow%3ARelease" rel="noopener nofollow ugc">this example</a>.</p>
